app/code/community/Bolt/Boltpay/controllers/ConfigurationController.php

```diff
@@ -136,7 +136,7 @@ protected function checkPublishableKey($key)
         try{
             $client = $this->boltHelper()->getApiClient();
             $response = $client->get($url,$headerInfo)->getStatusCode();
-            return (int)($response / 100) == 2;
+            return (int)round($response / 100) == 2;
```

Comment:
This is unrelated, unnecessary, and wrong.  This is not related to discount values and is checking HTTP response codes.  Please revert this back to the original form.